app/models/local_transaction_service.rb

```diff
@@ -1,41 +1,43 @@
 class LocalTransactionService
-  def self.config
-    @config ||= begin
-      filepath = Rails.root.join(Frontend.local_transactions_config)
-      YAML.safe_load(File.open(filepath))["services"]
-    end
-  end
+  attr_reader :lgsl, :devolved_administration_availability, :country_name
 
-  attr_reader :lgsl
-
-  def initialize(name, lgsl, country_name, local_authority_homepage)
+  def initialize(name, lgsl, country_name, local_authority_homepage, devolved_administration_availability)
     @name = name
     @lgsl = lgsl
     @country_name = country_name
     @local_authority_homepage = local_authority_homepage
-
-    @unavailable_content = LocalTransactionService.config.dig(
-      lgsl, "unavailable_in", @country_name
-    ) || {}
+    @devolved_administration_availability = devolved_administration_availability
   end
 
-  def body
-    @unavailable_content["body"]
+  def unavailable?
+    find_devolved_administration_availability[country_name] == "unavailable"
```

Comment:
If we had the presenter class available here I wonder if we could pass some of the work of this class that way.

For example this code could be:
```
def unavailable?
  @presenter.unavailable?(country_name)
end
```

and then in the presenter we have something like
```
def unavailabe?(country_name)
  key = country_name.downcase.gsub(" ", "_")
  @content_item.dig("#{key}_availability", "type") == "unavailable"
end
```
or similar - since we probably don't want this class to know too much about the content item details.

Reply:
As I think about this more I wonder whether we're going to need this LocalTransactionService class by moving the devolved_administration stuff into the Presenter. I think we're going to leave this class in a strange place by adapting it as most of it's concerns seem to be based around unavailable handling - with much of it becoming redundant by this work.

I'm imagining we can probably adapt the controller so it's like:

```
    if @publication.unavailable?(@country_name)
      render :unavailable_service
    elsif @publication.devolved_administration?(@country_name)
      render :devolved_administration_service
    else
      render :results
    end
```

And enable us to delete this class